sw/device/silicon_creator/rom/e2e/rom_e2e_ret_ram_init_test.c

```diff
@@ -0,0 +1,62 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/dif/dif_rstmgr.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/test_framework/check.h"
+#include "sw/device/lib/testing/test_framework/ottf_main.h"
+#include "sw/device/silicon_creator/lib/drivers/retention_sram.h"
+#include "sw/device/silicon_creator/lib/drivers/rstmgr.h"
+#include "sw/device/silicon_creator/lib/test_main.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+OTTF_DEFINE_TEST_CONFIG();
+
+// Variables of type `retention_sram_t` are static to reduce stack usage.
+static retention_sram_t ret;
+static uint64_t raw[sizeof(retention_sram_t) / sizeof(uint64_t)];
```

Comment:
nit: can you either (1) add an assert that `sizeof(retention_sram_t)` is a multiple of `sizeof(uint64_t)`, or (2) just add `+1` to the number of elements to make sure that we don't do OOB access.